expression/builtin_time_vec.go

```diff
@@ -979,11 +979,41 @@ func (b *builtinUnixTimestampDecSig) vecEvalDecimal(input *chunk.Chunk, result *
 }
 
 func (b *builtinPeriodAddSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalInt evals PERIOD_ADD(P,N).
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_period-add
 func (b *builtinPeriodAddSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	i64s := result.Int64s()
+	result.MergeNulls(buf)
+	ns := buf.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+
+		// this is non-vectorized implementation
```

Comment:
I mean this line.
https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L5451

Reply:
```suggestion
		// in MySQL, if p is invalid but n is NULL, the result is NULL, so we have to check if n is NULL first.
```